azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -12,11 +12,29 @@
 from azure.iot.device.common.evented_callback import EventedCallback
 from .abstract_provisioning_device_client import AbstractProvisioningDeviceClient
 from .abstract_provisioning_device_client import log_on_register_complete
-from .internal.polling_machine import PollingMachine
+from azure.iot.device.provisioning.pipeline import constant as dps_constant
+from .pipeline import exceptions as pipeline_exceptions
+from azure.iot.device import exceptions
+
 
 logger = logging.getLogger(__name__)
 
 
+def handle_result(callback):
```

Comment:
>handle_result [](start = 4, length = 13)

suggest renaming.  "handle" doesn't tell me what this function does.  maybe "wait_for_registration_completion" or something like that?